tools/querytee/proxy_test.go

```diff
@@ -68,11 +68,124 @@ func testRequestTransformer2(r *http.Request, body []byte, _ *spanlogger.SpanLog
 }
 
 func Test_NewProxy(t *testing.T) {
-	cfg := ProxyConfig{}
+	testCases := map[string]struct {
+		cfg           ProxyConfig
+		expectedError string
+	}{
+		"empty config": {
+			cfg: ProxyConfig{
+				SecondaryBackendsRequestProportion: 1.0,
+			},
+			expectedError: "at least 1 backend is required",
+		},
+		"single endpoint, preferred backend set and exists": {
+			cfg: ProxyConfig{
+				BackendEndpoints:                   "http://blah",
+				PreferredBackend:                   "blah",
+				SecondaryBackendsRequestProportion: 1.0,
+			},
+			expectedError: "",
+		},
+		"single endpoint with subdirectory and port, preferred backend set and exists": {
+			cfg: ProxyConfig{
+				BackendEndpoints:                   "http://blah.com:1234/some-sub-dir/and-another",
```

Comment:
(I think there is a limitation here with query-tee where if you have two backends on the same domain with different sub-dir endpoints they can't both exist.. not something I think we need to worry about for the time being though).

Reply:
Yep - that's always been the case, and I think is something we could fix separately later.